src/objects.h

```diff
@@ -215,6 +215,10 @@ class HeapObject : public Object {
     return HeapObject::cast(_at(HEADER_OFFSET));
   }
 
+  INLINE void set_forwarding_address(HeapObject* destination) {
+    _at_put(HEADER_OFFSET, destination);
```

Comment:
Add a comment to the `HEADER_OFFSET` explaining that it can contain the forward address.